projects/forms/src/lib/validators/has-future-date/has-future-date.validator.ts

```diff
@@ -0,0 +1,22 @@
+import { FormControl, ValidationErrors, ValidatorFn } from '@angular/forms';
+
+/**
+ * hasFutureDateValidator
+ *
+ * Validator function to ensure that the selected date is not in the future.
+ * If the date is in the future, it returns an error.
+ * @returns ValidationErrors if the date is in the future, otherwise null.
+ *
+ */
+export const hasFutureDateValidator = (): ValidatorFn => {
+	return (control: FormControl): ValidationErrors | null => {
+		if (!control.value) {
+			return null;
+		}
+
+		const currentDate = new Date();
+		const inputDate = new Date(control.value);
+
+		return inputDate <= currentDate ? null : { invalidDate: { valid: false } };
```

Comment:
Hier zou ik een andere error teruggeven dan `invalidDate`, want dit wordt al gebruikt door andere validators als ik het goed op heb en is eigenlijk niet descriptive genoeg. Ik zou dit bv `isFutureDate`.